### PR TITLE
Np 45343 add facet has file

### PR DESCRIPTION
**Pull request by EyobSikt**
Added facet HasFile: - associatedArtifacts.type=PublishedFile and associatedArtifacts.administrativeAgreement=false

### CODE REVIEW

File: search-commons/src/test/java/no/unit/nva/search/OpensearchTest.java [hunk 1]
```diff
@@ -519,6 +519,19 @@ void shouldQueryingFundingSuccessfully() throws InterruptedException, ApiGateway
...
+        void shouldQueryingHasFileSuccessfully() throws InterruptedException, ApiGatewayException {
+            addDocumentsToIndex("sample_publication_with_affiliations.json",
+                                "sample_publication_with_several_of_the_same_affiliation.json");
+
+            var query = queryWithTermAndAggregation(
+                "associatedArtifacts.type:\"PublishedFile\" AND associatedArtifacts.administrativeAgreement:\"false\"",
+                ApplicationConstants.RESOURCES_AGGREGATIONS);
+
+            var response = searchClient.searchWithSearchDocumentQuery(query, indexName);
+            assertThat(response.getHits(), hasSize(1));
```
Comment (sondrev) [inline search-commons/src/test/java/no/unit/nva/search/OpensearchTest.java:532]: Your testing search here - that you can search for the documents with the fields you have created aggregations for. Not that the returned response from opensearch actually returns the aggregation itself. Look into the function "assertAggregation" in opensearchtest.java

File: search-commons/src/main/java/no/unit/nva/search/constants/ApplicationConstants.java [hunk 2]
```diff
@@ -188,4 +190,18 @@ private static NestedAggregationBuilder generateLabelsAggregation(String object)
...
+
+    private static FilterAggregationBuilder generateHasFileAggregation() {
+
+        FilterAggregationBuilder typeFilterAggregation = AggregationBuilders.filter("associatedArtifacts.type",
+                                                                                    QueryBuilders.termQuery("associatedArtifacts.type.keyword", "PublishedFile"));
+
+        FilterAggregationBuilder adminAgreementFilterAggregation = AggregationBuilders.filter("associatedArtifacts"
+                                                                                + ".administrativeAgreement",
+                                                                                QueryBuilders.termQuery("associatedArtifacts.administrativeAgreement", false));
+
```
Comment (sondrev) [inline search-commons/src/main/java/no/unit/nva/search/constants/ApplicationConstants.java:202]: Use jsonPath and constants instead. This applied to  all lines in this method

QueryBuilders.termQuery(jsonPath(ASSOCIATED_ARTIFACTS,ADMINISTRATIVE_AGREEMENT, false));

**Review by sondrev — CHANGES_REQUESTED** [review]

File: search-commons/src/test/java/no/unit/nva/search/OpensearchTest.java (hunk 1 above)
Comment (EyobSikt) [reply search-commons/src/test/java/no/unit/nva/search/OpensearchTest.java]: Done.  But here no need to use assertAggregation as we do not have buckets. w e have only docCount.

File: search-commons/src/main/java/no/unit/nva/search/constants/ApplicationConstants.java [hunk 3]
```diff
@@ -191,17 +193,16 @@ private static String jsonPath(String... args) {
         return String.join(JSON_PATH_DELIMITER, args);
     }
 
-    private static FilterAggregationBuilder generateHasFileAggregation() {
+    private static NestedAggregationBuilder generateHasFileAggregation() {
 
-        FilterAggregationBuilder typeFilterAggregation = AggregationBuilders.filter("associatedArtifacts.type",
-                                                                                    QueryBuilders.termQuery("associatedArtifacts.type.keyword", "PublishedFile"));
+        FilterAggregationBuilder typeFilterAggregation = AggregationBuilders.filter(TYPE,
+                                                                                    QueryBuilders.termQuery(jsonPath(ASSOCIATED_ARTIFACTS, TYPE, KEYWORD), "PublishedFile"));
```
Comment (sondrev) [inline search-commons/src/main/java/no/unit/nva/search/constants/ApplicationConstants.java:199]: you forgot "PublishedFile"

Comment (EyobSikt) [reply search-commons/src/main/java/no/unit/nva/search/constants/ApplicationConstants.java]: Ok. Added it.

**Review by sondrev — APPROVED** [review]

**Pull request merged**